packages/standard-asset-apis/test/routers/KeyRouter-spec.ts

```diff
@@ -90,11 +90,187 @@ describe('KeyRouter', () => {
         expect(router.lookup(entity)).toEqual('FOO');
     });
 
+    it('should work with use 1 from the beginning and setting suffix for uppercase character', () => {
+        const router = new KeyRouter({
+            case: KeyRouterCaseOptions.lower,
+            use: 1,
+            from: KeyRouterFromOptions.beginning,
+            suffix_use: true,
+            suffix_upper: '--u'
+        });
+
+        const entity = new DataEntity({ foo: 'bar' });
+        entity.setKey('Rm9vQmFy');
+        expect(entity.getKey()).toEqual('Rm9vQmFy');
+        expect(router.lookup(entity)).toEqual('r--u');
+    });
+
+    it('should work with use 1 from the beginning and setting suffix for uppercase character and case upper', () => {
+        const router = new KeyRouter({
+            case: KeyRouterCaseOptions.upper,
+            use: 1,
+            from: KeyRouterFromOptions.beginning,
+            suffix_use: true,
+            suffix_upper: '--u'
+        });
+
+        const entity = new DataEntity({ foo: 'bar' });
+        entity.setKey('Rm9vQmFy');
+        expect(entity.getKey()).toEqual('Rm9vQmFy');
+        expect(router.lookup(entity)).toEqual('R--u');
```

Comment:
how is this equal to `R--u`? Since you have the caseFn called on the outer return value from the suffix method, should this not be `R--U`?